src/controllers/publish.ts

```diff
@@ -85,11 +87,39 @@ export const uploadFile = async (req: Request, res: Response, next: NextFunction
 
     if (req.method === 'POST') {
         try {
-            if (!req.file || req.file.mimetype !== 'text/csv') {
+            if (!req.file) {
                 throw new Error('errors.csv.invalid');
             }
-            logger.debug('File upload submitted...');
             const fileName = req.file.originalname;
+            if (req.file.mimetype === 'application/octet-stream') {
+                const ext = path.extname(req.file.originalname);
+                switch (ext) {
+                    case '.parquet':
+                        req.file.mimetype = 'application/vnd.apache.parquet';
+                        break;
+                    case '.json':
+                        req.file.mimetype = 'application/json';
+                        break;
+                    case '.xls':
+                    case '.xlsx':
+                        req.file.mimetype = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+                        break;
+                    case '.csv':
+                        req.file.mimetype = 'text/csv';
+                        break;
+                    default:
+                        throw new Error(`unsupported format ${ext}`);
+                }
+            } else if (req.file.mimetype === 'application/x-gzip') {
+                const ext = req.file.originalname.split('.').reverse()[1];
+                switch (ext) {
+                    case 'json':
+                    case 'csv':
+                        break;
+                    default:
+                        throw new Error(`unsupported format ${ext}`);
+                }
+            }
```

Comment:
I'd probably extract this to a util function but not a blocker.

Reply:
I'll do it with pagination.